File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DataGridView/Performance/VirtualModePage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;

namespace CompanionKit.Controls.DataGridView.Features
{
    public partial class VirtualModePage : UserControl
    {
      
        public VirtualModePage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the Click event of the mobjFillDGVButton control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjFillDGVButton_Click(object sender, EventArgs e)
        {
            //Clears all records if such exist
            if (mobjDGVUserDS.Users.Count > 0)
            {
                mobjDGVUserDS.Users.Clear();
            }
            // Fills up the DataGridView.
            for (int i = 1; i < 3000; i++)
            {
                mobjDGVUserDS.Users.AddUsersRow(string.Format("User{0}", i), string.Format("user{0}@gmail.com", i),
                                         string.Format("8-800-236589{0}", i), "Franklin",
                                         string.Format("10{0} Murfreeboro Rd.", i), "USA", "Tennessee", "37064");
            }
            //Creates temp Column array and fills with current values 
            DataGridViewColumn[] mobjTempCollection = new DataGridViewColumn[mobjDataGridView.Columns.Count];
            mobjDataGridView.Columns.CopyTo(mobjTempCollection, 0);
            //Temporary unbinding datasource
            mobjDataGridView.DataSource = null;
            //If DVG doesn't contain any column - fills with temp array  
            if (mobjDataGridView.Columns.Count == 0)
            {
                mobjDataGridView.Columns.AddRange(mobjTempCollection);
            }
            //Shows warning message with possibility to stop binding operation or continue it
            MessageBox.Show("Press 'Yes' to cancel this operation or press 'No' to continue (probably, page will fall)", "Page is not responding", MessageBoxButtons.YesNo, MessageBoxIcon.Warning, OnTimeout);
        }

        /// <summary>
        /// Handles the Click event of the mobjFillVDGVButton control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjFillVDGVButton_Click(object sender, EventArgs e)
        {
            //Clears all records if such exist
            if (mobjVDGVUserDS.Users.Count > 0)
            {
                mobjVDGVUserDS.Users.Clear();
            }
            // Fills up the VirtualDataGridView.
            for (int i = 1; i < 3000; i++)
            {
                mobjVDGVUserDS.Users.AddUsersRow(string.Format("User{0}", i), string.Format("user{0}@gmail.com", i),
                                         string.Format("8-800-236589{0}", i), "Franklin",
                                         string.Format("10{0} Murfreeboro Rd.", i), "USA", "Tennessee", "37064");
            }
        }

        /// <summary>
        /// Called when [timeout].
        /// </summary>
        /// <param name="sender">The sender.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void OnTimeout(object sender, EventArgs e)
        {
            //If 'Yes' button pressed - clear all added data  
            if (((MessageBoxWindow)sender).DialogResult == DialogResult.Yes)
            {
                mobjDGVUserDS = new UserDS();
                mobjDGVBindingSource.DataSource = mobjDGVUserDS;
                mobjDataGridView.DataSource = mobjDGVBindingSource; 
            }
            //If 'No' button pressed - bind back datasource
            else { mobjDataGridView.DataSource = mobjDGVBindingSource;}
        }
    }
}

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/DataGridView/Performance/VirtualModePage.vb
Namespace CompanionKit.Controls.DataGridView.Features

	Public Class VirtualModePage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

		End Sub

        ''' <summary>
        ''' Handles the Click event of the mobjFillDGVButton control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjFillDGVButton_Click(sender As Object, e As EventArgs) Handles mobjFillDGVButton.Click
            'Clears all records if such exist
            If mobjDGVUserDS.Users.Count > 0 Then
                mobjDGVUserDS.Users.Clear()
            End If
            ' Fills up the DataGridView.
            For i As Integer = 1 To 2999
                mobjDGVUserDS.Users.AddUsersRow(String.Format("User{0}", i), String.Format("user{0}@gmail.com", i), String.Format("8-800-236589{0}", i), "Franklin", String.Format("10{0} Murfreeboro Rd.", i), "USA", _
                 "Tennessee", "37064")
            Next
            'Creates temp Column array and fills with current values 
            Dim mobjTempCollection As DataGridViewColumn() = New DataGridViewColumn(mobjDataGridView.Columns.Count - 1) {}
            mobjDataGridView.Columns.CopyTo(mobjTempCollection, 0)
            'Temporary unbinding datasource
            mobjDataGridView.DataSource = Nothing
            'If DVG doesn't contain any column - fills with temp array  
            If mobjDataGridView.Columns.Count = 0 Then
                mobjDataGridView.Columns.AddRange(mobjTempCollection)
            End If
            'Shows warning message with possibility to stop binding operation or continue it
            MessageBox.Show("Press 'Yes' to cancel this operation or press 'No' to continue (probably, page will fall)", "Page is not responding", MessageBoxButtons.YesNo, MessageBoxIcon.Warning, AddressOf OnTimeout)
        End Sub

        ''' <summary>
        ''' Handles the Click event of the mobjFillVDGVButton control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub mobjFillVDGVButton_Click(sender As Object, e As EventArgs) Handles mobjFillVDGVButton.Click
            'Clears all records if such exist
            If mobjVDGVUserDS.Users.Count > 0 Then
                mobjVDGVUserDS.Users.Clear()
            End If
            ' Fills up the VirtualDataGridView.
            For i As Integer = 1 To 2999
                mobjVDGVUserDS.Users.AddUsersRow(String.Format("User{0}", i), String.Format("user{0}@gmail.com", i), String.Format("8-800-236589{0}", i), "Franklin", String.Format("10{0} Murfreeboro Rd.", i), "USA", _
                 "Tennessee", "37064")
            Next
        End Sub

        ''' <summary>
        ''' Called when [timeout].
        ''' </summary>
        ''' <param name="sender">The sender.</param>
        ''' <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        Private Sub OnTimeout(sender As Object, e As EventArgs)
            'If 'Yes' button pressed - clear all added data  
            If DirectCast(sender, MessageBoxWindow).DialogResult = DialogResult.Yes Then
                mobjDGVUserDS = New UserDS()
                mobjDGVBindingSource.DataSource = mobjDGVUserDS
                mobjDataGridView.DataSource = mobjDGVBindingSource
            Else
                'If 'No' button pressed - bind back datasource
                mobjDataGridView.DataSource = mobjDGVBindingSource
            End If
        End Sub
    End Class

End Namespace